Tests/SentryTests/Integrations/SessionReplay/SentrySessionReplayIntegrationTests.swift

```diff
@@ -68,6 +81,21 @@ class SentrySessionReplayIntegrationTests: XCTestCase {
         expect(SentrySDK.currentHub().trimmedInstalledIntegrationNames().count) == 1
         expect(SentryGlobalEventProcessor.shared().processors.count) == 1
     }
+    
+    func testWaitForNotificationWithNoWindow() {
+        uiApplication.windowsMock = nil
+        startSDK(sessionSampleRate: 1, errorSampleRate: 0)
+        
+        guard let sut = SentrySDK.currentHub().installedIntegrations().first as? SentrySessionReplayIntegration else {
+            fail("Did not installed replay integration")
+            return
+        }
+        
+        expect(Dynamic(sut).sessionReplay.asObject) == nil
+        uiApplication.windowsMock = [UIWindow()]
+        NotificationCenter.default.post(name: UIScene.didActivateNotification, object: nil)
```

Comment:
`m`: Please use the `TestNSNotificationCenterWrapper` if possible because the NotificationCenter lead to flaky tests in the past, see:

https://github.com/getsentry/sentry-cocoa/blob/c8dbe735adc49b67a92bacb74930d826a23ca7ee/Sources/Sentry/include/SentryNSNotificationCenterWrapper.h#L5-L13

Reply:
I tried, and it was a nightmare. I prefer to leave it like this.

Reply:
Why was it a nightmare? What was the problem?

Reply:
Many SIGNCRASH without a reason or proper stack trace.

The SessionReplayIntegration retrieves the notification wrapper from the Dependency Container, which is used by many other classes. After attempting to change the wrapper within the container, other objects were trying to remove observers from the wrapper during clean up, and the Invocation class was trying to access disposed objects.

Reply:
> The SessionReplayIntegration retrieves the notification wrapper from the Dependency Container, which is used by many other classes.

That's a major reason why I always advocate for passing in dependencies via constructors and not using the dependency container, which is global state and has side effects. Passing in dependencies via the constructor minimizes side effects. Anyways, we don't need to fix it in this PR.

Reply:
Integrations cannot have objects passed in the init function.
The problem here is with `Invocation` class, not the `SentryDependencyContainer`

Reply:
> Integrations cannot have objects passed in the init function.

Yes, I know, but most likely other classes can.

Reply:
> The problem here is the Invocation class, not SentryDependencyContainer

I think it's a mixture of both. So yes, the invocation class should use weak references in that case, and we should less of the `SentryDependencyContainer`. IMO.